monitoring/scd_qualifier/flight_volumes_generator.py

```diff
@@ -0,0 +1,307 @@
+                volume_generation_rule = VolumeGenerationRule(intersect_altitude= 0, intersect_time=0, expected_result = 1)
+                if path_index == 0:
+                    volume_generation_rule.is_control = 1
+            else:
+                # intersect in time and / or intersect in altitude 
+                volume_generation_rule = random.choice([VolumeGenerationRule(intersect_altitude=1, intersect_time= 0, expected_result = 1),VolumeGenerationRule(intersect_altitude=1, intersect_time= 1, expected_result = 0),VolumeGenerationRule(intersect_altitude=0, intersect_time= 1, expected_result = 1)])
+            all_volume_rules.append(volume_generation_rule)
+        return all_volume_rules
+
+    def generate_astm_4d_volumes(self,raw_geometries:List[GeometryPayload],rules : List[GeometryPayload], altitude_of_ground_level_wgs_84 :int) -> List[Volume4D]:
+        ''' A method to generate ASTM specified Volume 4D payloads to submit to the system to be tested.  '''
+        all_volume_4d :List[Volume4D] = []
+        for path_index, raw_geometry in enumerate(raw_geometries): 
+            volume_generation_rule = rules[path_index]
+            
+            flight_volume_3d = self.convert_geometry_to_volume(flight_geometry = raw_geometry.geometry, volume_generation_rule = volume_generation_rule, altitude_of_ground_level_wgs_84 = altitude_of_ground_level_wgs_84)
+            flight_volume_4d = self.transform_3d_volume_to_astm_4d(volume_3d = flight_volume_3d, volume_generation_rule = volume_generation_rule)
+            all_volume_4d.append(flight_volume_4d)
+            
+        return all_volume_4d
+
+
+class SCDFlightPathVolumeWriter():
+    """ A class to write raw Flight Paths and volumes to disk so that they can be examined / used in other software """
+
+    def __init__(self, raw_geometries:List[GeometryPayload],  flight_volumes: List[Volume4D], all_rules: List[VolumeGenerationRule],country_code='che')-> None:
+        '''
+        A method to write flight paths, volumes and volume generation rules to disk for review / submission to the test harness. All data is written in the `test_definitions` directory which is created if it does not exist.
+        
+        '''
+        self.country_code = country_code
+        self.output_directory = Path('test_definitions', self.country_code)        
+        # Create test_definition directory if it does not exist        
+        self.output_directory.mkdir(parents=True, exist_ok=True)
+        self.output_subdirectories = (Path(self.output_directory, 'astm_4d_volumes'), Path(self.output_directory, '4d_flight_geometries'),Path(self.output_directory, 'scd_rules'),)
+        for output_subdirectory in self.output_subdirectories:
+            output_subdirectory.mkdir(parents=True, exist_ok=True)
+        self.raw_geometries = raw_geometries
+        self.flight_volumes = flight_volumes
+        self.all_rules = all_rules
+
+    def write_4d_flight_geometries(self) -> None:
```

Comment:
The main reason this was written, was because of the manual checkout, one of my goals was to build the manual check out data with automated tools so that we start doing things simultaneously. This is not really required for fully automated tests. I will remove this section of code.